rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/TextDocumentState.java

```diff
@@ -158,22 +194,58 @@ private void parseIfNotParsing() {
             if (parsing.compareAndSet(false, true)) {
                 parser
                     .apply(location, content)
-                    .thenApply(t -> new Versioned<>(version, t))
-                    .whenComplete((t, error) -> {
-                        if (t != null) {
-                            var errors = RECOVERY.findAllErrors(t.get());
-                            if (errors.isEmpty()) {
-                                Versioned.replaceIfNewer(lastWithoutErrors, t);
+                    .whenComplete((t, e) -> {
+
+                        // Prepare result values for futures
+                        var tree = new Versioned<>(version, t);
```

Comment:
As a `<tree,diagnostics>` pair seem to belong together (not suprisingly), would ik make sense to have one `Versioned` object with a version and a pair of `<tree,diagnostics>`?